### ISSUE TITLE
Add example to interactively manipulate structure/molecule

### ISSUE THREAD

**Issue by janosh** [materialsproject/crystaltoolkit#309]
Just posting a Slack conversation with @exenGT so we don't forget about it. And maybe to get some thoughts from @mkhorton.

> 
> @exenGT I'm wondering if there are currently functionalities that allow direct manipulation of atoms / molecules in GUI
> For example, I'm thinking of a functionality where you can translate or rotate specific parts of a molecule
> If it's not there yet, I think I would like to add it if I have enough time
> 
> @janosh It's not currently possible to manipulate struct/mol in real time but that could definitely become an example app
> you would have to hook up a text input to control bond distance/angle with the `StructureMoleculeComponent`
> definitely doable
> 
> @exenGT Yep, I think it's a very useful thing to add as I use these a lot, but currently only via python
> Hence it's difficult to directly see the effect in real time

**Comment by mkhorton** [comment]
Yes, currently the only modifications can be done via pymatgen `Transformation` classes. So some of this might be implementable within the current framework, and we should make sure to support all the transformations we can.

However, this would not be as user friendly as simply manipulating the atoms directly in the GUI (this kind of feature I think is too hard to implement with the current team size). But if anyone has ideas for how to implement and wants to take it on, I think that would be excellent.
